modular_skyrat/modules/paycheck_rations/code/quirk.dm

```diff
@@ -0,0 +1,54 @@
+/datum/quirk/ration_system
+	name = "Ration Ticket Receiver"
+	desc = "Due to some circumstance of your life, you have enrolled in the ration tickets program, \
+		which will halve all of your paychecks in exchange for granting you ration tickets, which can be \
+		redeemed at a cargo console for food and other items."
+	icon = FA_ICON_DONATE
+	quirk_flags = QUIRK_HUMAN_ONLY|QUIRK_HIDE_FROM_SCAN
+	value = 0
+	medical_record_text = "Alas, the patient struggled to scrape together enough money to pay the checkup bill."
+	hardcore_value = 0
+
+/datum/quirk/ration_system/add_unique(client/client_source)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	if(!human_holder.account_id)
+		return
+	var/datum/bank_account/account = SSeconomy.bank_accounts_by_id["[human_holder.account_id]"]
+	account.payday_modifier = 0.5
+	account.gets_ration_tickets = TRUE
+	to_chat(client_source.mob, span_warning("You remember, you can turn in the ration tickets your paychecks give you to a cargo console for things! Neat!"))
+
+// Edits to bank accounts to make the above possible
+
+/datum/bank_account
+	/// Variable that tracks if this account should get ration tickets each payday
+	var/gets_ration_tickets = FALSE
+	/// Tracks if the last ticket we got was for luxury items
+	var/last_ticket_luxury = FALSE
+
+/datum/bank_account/payday(amount_of_paychecks, free = FALSE)
+	if(!(..() && gets_ration_tickets))
+		return
+	make_ration_ticket()
```

Comment:
If you want it to be 1:1 with paychecks received: 
```suggestion
	for(var/i in 1 to amount_of_paychecks)
		make_ration_ticket()
```

Reply:
Nah I think it should just be a single one per payday, just to be safe.